src/drivers/CensusApiRetryDriver.ts

```diff
@@ -1,51 +1,72 @@
 import {Rest} from 'ps2census';
 import {promiseTimeout} from '../utils/PromiseTimeout';
 import {Logger} from '@nestjs/common';
+import MetricsHandler from '../handlers/MetricsHandler';
+import {METRICS_NAMES} from '../modules/metrics/MetricsConstants';
+import {ConfigService} from '@nestjs/config';
 
 // This class exists as Census occasionally sends us invalid responses, and we must retry them.
 export class CensusApiRetryDriver<T extends Rest.CollectionNames> {
     private static readonly logger = new Logger('CensusApiRetryDriver');
 
-    // The below with the combination gives Census 30 seconds to recover before we abort a query.
-    private readonly retryLimit = 12;
+    // The below with the combination gives Census 120 seconds to recover before we abort a query.
+    private readonly retryLimit = 4;
     private readonly delayTime = 2500;
+    private readonly timeoutTime: number; // Maximum Census tolerance 30s
+    private readonly endpoint: string;
 
     constructor(
         private readonly query: Rest.GetQuery<T>,
         private readonly filter: Rest.Conditions<T>,
         private readonly caller: string,
-    ) {}
+        private readonly metricsHandler: MetricsHandler,
+        config: ConfigService,
+    ) {
+        this.timeoutTime = config.get('census.timeout'); // Maximum Census tolerance
+        this.endpoint = `/${this.query.collection}`;
+    }
 
     public async try(attempts = 0): Promise<Rest.CensusResponse<Rest.Format<T>> | undefined> {
```

Comment:
As we talked about, the return is not the format that the query returns, but rather the base format of a collection without the joins.

Reply:
Honestly, I don't know how to resolve this. I need more ELI5 unfortunately!